base/acme/src/main/java/org/dogtagpki/acme/backend/ACMEBackend.java

```diff
@@ -30,10 +34,54 @@ public void init() throws Exception {
     public void close() throws Exception {
     }
 
-    public BigInteger issueCertificate(String csr) throws Exception {
+    /**
+     * This method generates a unique ID for a certificate.
+     *
+     * By default this method will return the base64-encoded serial number
+     * of the certificate. This method may be overridden to generate a backend-
+     * specific unique ID for the certificate.
+     *
+     * @param certData Certificate data in byte array.
+     * @return Unique ID for the certificate.
+     * @throws Exception
+     */
+    public String getCertificateID(byte[] certData) throws Exception {
+
+        try (ByteArrayInputStream is = new ByteArrayInputStream(certData)) {
+
+            CertificateFactory cf = CertificateFactory.getInstance("X.509");
```

Comment:
Should I create a `CertificateFactory` for JSS? We don't actually have one yet.

Reply:
That's be nice. Ideally PKI should use Java standard API whenever possible.